backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractProvider.java

```diff
@@ -288,20 +271,9 @@ public Stream<FlatrateTermId> streamModularPurchaseContractsForShipmentLine(@Non
 			return Stream.empty();
 		}
 
-		final YearAndCalendarId yearAndCalendarId = YearAndCalendarId.ofRepoId(harvestingYearId, harvestingCalendarId);
-		final List<ModularContractSettings> settings = modularContractSettingsBL.getSettingsByQuery(ModularContractSettingsQuery.builder()
-				.processedProductId(productId)
-				.yearAndCalendarId(yearAndCalendarId)
-				.soTrx(SOTrx.PURCHASE)
-				.checkHasCompletedModularCondition(true)
-				.build());
-
-		final ProductId settingsProductId = CollectionUtils.extractSingleElementOrDefault(settings, ModularContractSettings::getRawProductId, null);
-		final ProductId productIdToUse = CoalesceUtil.coalesceNotNull(settingsProductId, productId);
-
 		final ModularFlatrateTermQuery query = ModularFlatrateTermQuery.builder()
 				.bPartnerId(warehouseBL.getBPartnerId(warehouseId))
-				.productId(productIdToUse)
+				.productId(productId)
```

Comment:
I think we don't need this, because of `de.metas.contracts.impl.FlatrateDAO#buildModularContractSettingsQueryFilter`